lib/src/equatable_utils.dart

```diff
@@ -48,10 +46,8 @@ int _combine(int hash, dynamic object) {
     });
     return hash;
   }
-  if (object is Set && object is! SplayTreeSet) {
-    // this is needed to have a consistent iteration order so that the produced
-    // hash is consistent independently of the Set insertion order
-    object = SplayTreeSet<dynamic>.from(object);
```

Comment:
We just care about the sort order and SplayTreeSet requires objects to be Comparable so this PR simplifies things by just sorting the set similar to the `Map` implementation.